lib/block_view_types.go

```diff
@@ -4804,6 +4912,11 @@ func DecodeMapStringUint64(rr *bytes.Reader) (map[string]uint64, error) {
 	return nil, nil
 }
 
+// EncodeUint256 is useful for space-efficient encoding of uint256s.
+// It does not guarantee fixed-width encoding, so should not be used
+// in BadgerDB keys. Use EncodeOptionalUint256 instead, which does
+// guarantee fixed-width encoding. Both EncodeUint256 and
+// EncodeOptionalUint256 can handle nil inputs.
 func EncodeUint256(number *uint256.Int) []byte {
```

Comment:
This is confusing because every other EncodeUintX function is fixed-width, which means we're guaranteed someone is going to produce a bug where they assumed it's fixed-width. I changed the name to VariableEncodeUint256 (and decode).